src/main/java/com/google/sps/servlets/RoutesStoring.java

```diff
@@ -79,14 +80,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     try {
       Entity userEntity = datastore.get(userKey);
       // Create new route entity and make it child of the user.
-      Entity routeEntity = new Entity("Route", userEntity.getKey());
+      Entity routeEntity = new Entity("Route");
 
       routeEntity.setProperty("routeName", routeName);
       routeEntity.setProperty("isPublic", isPublic);
       routeEntity.setProperty("startHour", startHour);
       routeEntity.setProperty("startMinute", startMinute);
 
       datastore.put(routeEntity);
+      for (long userId : editorsArray) {
+        Entity linkEntity = new Entity("Link", KeyFactory.createKey("User", userId));
+        linkEntity.setProperty("routeId", routeEntity.getKey().getId());
+        linkEntity.setProperty("type", 2);
+        // Add entity for editor.
+        datastore.put(linkEntity);
+      }
+
+      Entity linkEntity = new Entity("Link", userEntity.getKey());
```

Comment:
nit: "Link" is not very descriptive, does not match the associated java class name, and might be confusing if you add addtional links between other entities in future.